inspirehep/modules/orcid/utils.py

```diff
@@ -209,3 +215,35 @@ def canonicalize_xml_element(element):
         exclusive=True,
     )
     return output_stream.getvalue()
+
+
+def get_literature_recids_for_orcid(orcid):
+    """Return the Literature recids that were claimed by an ORCiD.
+
+    We record the fact that the Author record X has claimed the Literature
+    record Y by storing in Y an author object with a ``$ref`` pointing to X
+    and the key ``curated_relation`` set to ``True``. Therefore this method
+    first searches the DB for the Author records for the one containing the
+    given ORCiD, and then uses its recid to search in ES for the Literature
+    records that satisfy the above property.
+
+    Args:
+        orcid (str): the ORCiD.
+
+    Return:
+        list(int): the recids of the Literature records that were claimed
+        by that ORCiD.
+
+    """
+    orcid_object = '[{"schema": "ORCID", "value": "%s"}]' % orcid
+    author_recid = db.session.query(PersistentIdentifier.pid_value)\
+                             .filter(PersistentIdentifier.object_uuid == RecordMetadata.id)\
+                             .filter(PersistentIdentifier.pid_type == 'aut')\
+                             .filter(type_coerce(RecordMetadata.json, JSONB)['ids'].contains(orcid_object))\
+                             .one().pid_value
```

Comment:
I want to share my thought about this query. I come from the Django ORM where joins are never explicit, so I often think in terms of 2 sequential queries rather than a join.

So my question is what is more performant between a join query and 2 sequential queries. Or if this question even makes sense.

Your query translates to:
```sql
SELECT pidstore_pid.pid_value AS pidstore_pid_pid_value
FROM pidstore_pid, records_metadata
WHERE pidstore_pid.object_uuid = records_metadata.id AND pidstore_pid.pid_type = %(pid_type_1)s AND ((records_metadata.json -> %(param_1)s)) @> %(param_2)s
```

While 2 sequential queries like:
```python
record_ids = db.session.query(RecordMetadata.id)\
                    .filter(type_coerce(RecordMetadata.json, JSONB)['ids'].contains(orcid_object)).all()
author_recid = db.session.query(PersistentIdentifier.pid_value)\
                         .filter(PersistentIdentifier.object_uuid.in_(record_ids))\
                         .filter(PersistentIdentifier.pid_type == 'aut').one()
```
translates to:
```sql
SELECT records_metadata.id AS records_metadata_id
FROM records_metadata
WHERE ((records_metadata.json -> %(param_1)s)) @> %(param_2)s

SELECT pidstore_pid.pid_value AS pidstore_pid_pid_value
FROM pidstore_pid
WHERE pidstore_pid.object_uuid IN (%(object_uuid_1)s) AND pidstore_pid.pid_type = %(pid_type_1)s
```

Reply:
You just made me realize an important point: there's an index on `(object_type, object_uuid)`, which is not used when joining on just `object_uuid`:
```
inspirehep=# EXPLAIN SELECT * FROM pidstore_pid WHERE object_uuid='b70bd0bb-4ea9-4b8b-a56c-a17910e27022';
                               QUERY PLAN                               
------------------------------------------------------------------------
 Seq Scan on pidstore_pid  (cost=0.00..11.50 rows=1 width=638)
   Filter: (object_uuid = 'b70bd0bb-4ea9-4b8b-a56c-a17910e27022'::uuid)
(2 rows)

inspirehep=# EXPLAIN SELECT * FROM pidstore_pid WHERE object_type='rec' AND object_uuid='b70bd0bb-4ea9-4b8b-a56c-a17910e27022';
                                                      QUERY PLAN                                                      
----------------------------------------------------------------------------------------------------------------------
 Index Scan using idx_object on pidstore_pid  (cost=0.14..8.16 rows=1 width=638)
   Index Cond: (((object_type)::text = 'rec'::text) AND (object_uuid = 'b70bd0bb-4ea9-4b8b-a56c-a17910e27022'::uuid))
(2 rows)
```

Reply:
Another possibility is to also filter on `object_type`, and then one can use the index. Something smarter would be to declare the index on `(object_uuid, object_type)`, so that it could be used for either!